cpp/src/arrow/csv/parser.cc

```diff
@@ -337,6 +340,9 @@ class BlockParserImpl {
     if (UseBulkFilter) {
       const char* bulk_end = RunBulkFilter(parsed_writer, data, data_end, bulk_filter);
       if (ARROW_PREDICT_FALSE(bulk_end == nullptr)) {
+        if (is_final) {
+          data = data_end;
+        }
```

Comment:
Is it an error if this happens in a quoted field?

Reply:
Hmm... I don't *think* so.